tests/integration/ha/test_ha.py

```diff
@@ -4,30 +4,68 @@
 
 import asyncio
 import logging
+import time
 
 import pytest
-from helpers import (
+from pytest_operator.plugin import OpsTest
+
+from integration.ha.continuous_writes import ContinuousWrites
+from integration.ha.ha_helpers import (
+    get_topic_leader,
+    get_topic_offsets,
+    patch_restart_delay,
+    remove_restart_delay,
+    send_control_signal,
+)
+from integration.helpers import (
     APP_NAME,
+    DUMMY_NAME,
     REL_NAME_ADMIN,
     ZK_NAME,
     check_logs,
     produce_and_check_logs,
 )
-from pytest_operator.plugin import OpsTest
 
 logger = logging.getLogger(__name__)
 
 
-DUMMY_NAME = "app"
+@pytest.fixture()
+async def c_writes(ops_test: OpsTest):
+    """Creates instance of the ContinuousWrites."""
+    app = APP_NAME
+    return ContinuousWrites(ops_test, app)
+
+
+@pytest.fixture()
+async def c_writes_runner(ops_test: OpsTest, c_writes: ContinuousWrites):
+    """Starts continuous write operations and clears writes at the end of the test."""
+    c_writes.start()
+    yield
+    c_writes.clear()
+    logger.info("\n\n\n\nThe writes have been cleared.\n\n\n\n")
+
+
+@pytest.fixture()
+async def restart_delay(ops_test: OpsTest):
+    for unit in ops_test.model.applications[APP_NAME].units:
+        await patch_restart_delay(ops_test=ops_test, unit_name=unit.name, delay=5)
+    yield
+    for unit in ops_test.model.applications[APP_NAME].units:
+        await remove_restart_delay(ops_test=ops_test, unit_name=unit.name)
```

Comment:
**praise** I like this patter to integrate continuous write and restart delay as fixtures! well done!